### ISSUE TITLE
Make KeyResolver and Formatter public

### ISSUE THREAD

**Issue by syhily** [mybatis/guice#96]
So the user like me could implement custom JdbcHelper.

**Comment by christianpoitras** [comment]
Hi, don't think making those classes public will help you achieve what you want. I'll look for a better solution to your problem.

**Comment by simonetripodi** [comment]
`JdbcHelper` is an _Enumeration_, changing `KeyResolver` and `Formatter` modifier won't help you implement a custom `JdbcHelper`.
What I'd suggest is contributing the missing helper for the interested driver

**Comment by christianpoitras** [comment]
Hi @simonetripodi !
What is interesting is allowing to use a different Formatter or KeyResolver rather than changing JdbcHelper. But I suspect that making those classes public will not allow people to replace the implementation easily.

**Comment by christianpoitras** [comment]
And as you said, if a user just wants an additional driver, they can just submit the change and I will gladly add the new values for the next release.

**Comment by syhily** [comment]
I should clarify my requirements. I just implement the JdbcHelper because I want to make mysql using slf4j. It would need to make jdbc url like `jdbc:mysql://${JDBC.host}:${JDBC.port}/${JDBC.schema}?autoReconnect=true&useSSL=false&logger=com.mysql.jdbc.log.Slf4JLogger&profileSQL=true`

So, it would help me a lot if the class `Formatter` be public. I could just write some code like below.
```
String urlTemplate = "jdbc:mysql://${JDBC.host}:${JDBC.port}/${JDBC.schema}?autoReconnect=true&useSSL=false&logger=com.mysql.jdbc.log.Slf4JLogger&profileSQL=true"
binder.bind(Key.get(String.class, named("JDBC.url"))).toProvider(guicify(new Formatter(urlTemplate)));
```

**Comment by christianpoitras** [comment]
I have a strong dislike to make Formatter public because of its name.

I will rename it JdbcUrlAntFormatter and then make it public. That way it will be clear that the class it not meant to be used for other purposes.

**Comment by syhily** [comment]
Tks, this helps a lot. 👍

**Comment by christianpoitras** [comment]
Fixed in commit f997dbc181216eb879dc7fb51f436cbfb1c1cbf0.
